grakn-kb/src/main/java/ai/grakn/kb/internal/concept/SchemaConceptImpl.java

```diff
@@ -135,24 +137,58 @@ public T sup() {
         return cachedSuperType.get();
     }
 
+
+    /**
+     *
+     * @return The supertypes of this
+     * TODO: add option to return unfiltered supertypes
+     */
+    @Override
+    public Stream<T> sups() {
+        return this.filterSuperSet(this.superSet());
+    }
+
     /**
      *
      * @return All outgoing sub parents including itself
      */
     public Stream<T> superSet() {
         Set<T> superSet= new HashSet<>();
+
         superSet.add(getThis());
         T superParent = sup();
 
         while(superParent != null && !Schema.MetaSchema.THING.getLabel().equals(superParent.getLabel())){
             superSet.add(superParent);
+
             //noinspection unchecked
             superParent = (T) superParent.sup();
         }
 
         return superSet.stream();
     }
 
+
+    /**
+     * Filters the supertypes not to return schema nodes
+     * @param superSet
+     * @return the filtered supertypes
+     */
+    public Stream<T> filterSuperSet(Stream<T> superSet) {
+        Set<T> filtered = new HashSet<>();
+        superSet.forEach((node) -> {
```

Comment:
I think you might be able to simplify this to the following:

```
return superSet.filter(concept -> !MetaSchema.isMetaLabel(concept.getLabel()));
```

Reply:
That is nice. I love using lambdas. I did not manage to find method `isMetaLabel()` Thank you for pointing this out to me. (by the way it is `!Schema.MetaSchema`)